src/redisearch_rs/inverted_index/src/lib.rs

```diff
@@ -395,18 +379,56 @@ impl<'a> IntoIterator for RSAggregateResult<'a> {
 #[derive(Debug, PartialEq)]
 pub struct RSVirtualResult;
 
+/// Repeats the [`RSResultData`] definition since `bitflags` does not support enum with values or
+/// enums with lifetimes currently.
```

Comment:
```suggestion
/// A C-style discriminant for [`RSResultData`].
///
/// # Implementation notes
///
/// We need a standalone C-style discriminant to get `bitflags` to generate a 
/// dedicated bitmask type. Unfortunately, we can't apply `#[bitflags]` directly
/// on [`RSResultData`] since `bitflags` doesn't support enum with data in
/// their variants, nor lifetime parameters.
```